lib/runtime/modules/array.js

```diff
@@ -28,7 +38,116 @@ var array = {
+
+        return array.join(joiner);
+    },
+    lastIndexOf: function(array, searchValue) {
+        if (!values.isArray(array)) {
+            throw errors.typeErrorFunction('Array.lastIndexOf', 'array', array);
+        }
+
+        var index = -1;
+
+        for (var i = array.length - 1; i > 0; i--) {
+            if (values.equal(array[i],searchValue)) {
+                index = i;
+                break;
+            }
+        }
+
+        return index;
+    },
+    pop: function(array) {
+        if (!values.isArray(array)) {
+            throw errors.typeErrorFunction('Array.pop', 'array', array);
+        }
+
+        return array.pop();
+    },
+    push: function(array) {
+        if (!values.isArray(array)) {
+            throw errors.typeErrorFunction('Array.push', 'array', array);
+        }
+
+        var pushArgs = _.toArray(arguments).slice(1);
+
+        return array.push.apply(array, pushArgs);
+    },
+    reverse: function(array) {
+        if (!values.isArray(array)) {
+            throw errors.typeErrorFunction('Array.reverse', 'array', array);
+        }
+
+        return array.reverse();
+    },
+    shift: function(array) {
+        if (!values.isArray(array)) {
+            throw errors.typeErrorFunction('Array.shift', 'array', array);
+        }
+
+        return array.shift();
+    },
+    slice: function(array, begin, end) {
+        if (!values.isArray(array)) {
+            throw errors.typeErrorFunction('Array.slice', 'array', array);
+        }
+        if (begin !== undefined && !values.isNumber(begin)) {
+            throw errors.typeErrorFunction('Array.slice', 'number', begin);
+        }
+        if (end !== undefined && !values.isNumber(end)) {
+            throw errors.typeErrorFunction('Array.slice', 'number', end);
+        }
+
+        return array.slice(begin, end);
+    },
+    sort: function(array) {
+        if (!values.isArray(array)) {
+            throw errors.typeErrorFunction('Array.sort', 'array', array);
+        }
+        return array.sort(function(a, b) {
+            return values.compare(a, b);
+        });
+    },
+    splice: function(array) {
+        if (!values.isArray(array)) {
+            throw errors.typeErrorFunction('Array.splice', 'array', array);
+        }
+        var begin = arguments[1];
+        if (begin !== undefined && !values.isNumber(begin)) {
+            throw errors.typeErrorFunction('Array.splice', 'number', begin);
+        }
+        var deleteCount = arguments[2];
+        if (deleteCount !== undefined && !values.isNumber(deleteCount)) {
+            throw errors.typeErrorFunction('Array.splice', 'number', deleteCount);
+        }
+
+        var spliceArgs = _.toArray(arguments).slice(1);
+        return array.splice.apply(array, spliceArgs);
+    },
+    toString: function(array) {
+        if (!values.isArray(array)) {
+            throw errors.typeErrorFunction('Array.toString', 'array', array);
+        }
+
+        return Array.toString(array);
```

Comment:
Do we want to use native representation? Maybe `values.toString` would be better (similarly for `Object`).